### PR TITLE
Us6

### PR DESCRIPTION
**Pull request by Leviron**

### CODE REVIEW

File: components/Form/Edit.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import { useState, useEffect } from "react";
+import { FormContainer, FormButton } from "./styles-form";
+import { Heading } from "./styles-form";
+import { useRouter } from "next/router";
+
+export default function EditRecipe() {
```
Comment (Roland-Hufnagel) [inline components/Form/Edit.js]: Please keep in mind: A component should be as dump as can be. That means: You do a lot of calculations here in the component. Thats not the right place. It's a lot better to do that in the corresponding page (pages/editpage/]id].js in your case)
There you should get the id from the router.query. There you should define  a handleSubmit-function. Then call the Component and pass the recipe and the handler-function as props.

File: components/Form/Edit.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+  const handleIngredientChange = (index, field, event) => {
+    const updatedIngredients = [...recipe.ingredients];
+    updatedIngredients[index][field] = event.target.value;
+
+    setRecipe({
+      ...recipe,
+      ingredients: updatedIngredients,
+    });
+  };
```
Comment (Roland-Hufnagel) [inline components/Form/Edit.js:46]: I don't know why you have that much different handlers here. I think it is enough to have the handleSubmit and there get the formdata and fetch the api/recipes/${id}.

File: pages/editpage/[id].js [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import CreateRecipe from "../../components/Form";
+import EditRecipe from "../../components/Form/Edit";
 
-export default function CreatePage() {
-  return <CreateRecipe />;
+export default function EditPage() {
+  return <EditRecipe />;
```
Comment (Roland-Hufnagel) [inline pages/editpage/[id].js]: As I mentioned above, here you should do the fetch and define a submithandler and pass it as props to the ``<EditRecipe>``-component.

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hi Ron, I left some comments. Please have a look and try to refactor your code.

**Comment by Leviron** [comment]
hello Roland, please have a look. I have implemented all your requests

File: components/Form/Edit.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    steps: "",
+    timers: "",
+  });
+
+  useEffect(() => {
+    if (id) {
+      fetchRecipe(id);
+    }
+  }, [id]);
+
```
Comment (florianblaschke) [inline components/Form/Edit.js]: I would also recommend you put the fetchRecipe out of the component and put it into the editpage[id].js

If I understood Roland correctly, that is also what he meant.

File: pages/editpage/[id].js [hunk 5]
```diff
@@ -1,5 +1,30 @@
...
+import EditRecipe from "../../components/Form/Edit";
 
-export default function CreatePage() {
-  return <CreateRecipe />;
+export default function EditRecipePage() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const handleSubmit = async (updatedRecipe) => {
+    try {
```
Comment (florianblaschke) [inline pages/editpage/[id].js]: I dont think you need a try/catch block here. You already have it in the API. If the fetch fails, you will fall into !response.ok

**Review by florianblaschke — COMMENTED** [review]
Form/Edit is still a pretty smart component. It works, but as soon as you need some data on top level you will have to restructure even more, e.g. the recipe state.

I would put all the handlings into the editpage.js but at least the fetch, IMHO, should be there

File: pages/editpage/[id].js (hunk 5 above)
Comment (Leviron) [reply pages/editpage/[id].js]: I removed the try/catch block

**Comment by Leviron** [comment]
Hi, please have a look. I have implemented all your requests

File: pages/editpage/[id].js [hunk 6]
```diff
@@ -1,5 +1,55 @@
...
+  const [initialRecipe, setInitialRecipe] = useState(null);
+
+  useEffect(() => {
+    async function fetchInitialRecipe() {
+      const recipe = await fetchRecipe(id);
+      setInitialRecipe(recipe);
+    }
+
+    fetchInitialRecipe();
+  }, [id]);
```
Comment (onemanwenttomow) [inline pages/editpage/[id].js]: Seeing as you have `swr` installed in this project, i think it makes sense to use that to fetch the data. 

```suggestion
 const { data: initialRecipe, isLoading } = useSWR(`/api/recipes/${id}`);
```

We would need to rename the destructured `data` to be `initialRecipe` and we can also get access to a `isLoading` state from `swr`.

File: pages/editpage/[id].js [hunk 7]
```diff
@@ -1,5 +1,55 @@
...
+async function fetchRecipe(id) {
+  try {
+    const response = await fetch(`/api/recipes/${id}`);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error fetching recipe:", error);
+    return null;
+  }
+}
```
Comment (onemanwenttomow) [inline pages/editpage/[id].js]:
```suggestion

```

If we decide to refactor to use `swr` then we can remove this.

File: pages/editpage/[id].js [hunk 8]
```diff
@@ -1,5 +1,55 @@
...
+
+    if (!response.ok) {
+      window.alert("Problem updating the recipe!");
+      return;
+    }
+
+    router.push("/");
+  };
+
+  if (initialRecipe === null) {
```
Comment (onemanwenttomow) [inline pages/editpage/[id].js]: If we decide to refactor to use `swr` then we can do the following. 

```suggestion
  if (isLoading) {
```

File: pages/editpage/[id].js [hunk 9]
```diff
@@ -1,5 +1,55 @@
...
+    console.error("Error fetching recipe:", error);
+    return null;
+  }
+}
+
+export default function EditRecipePage() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [initialRecipe, setInitialRecipe] = useState(null);
```
Comment (onemanwenttomow) [inline pages/editpage/[id].js]: If we decide to refactor to use `swr` then we no longer need this line.

File: pages/editpage/[id].js [hunk 10]
```diff
@@ -1,5 +1,55 @@
-import CreateRecipe from "../../components/Form";
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
```
Comment (onemanwenttomow) [inline pages/editpage/[id].js]: If we decide to refactor to use `swr` then we no longer need this line. 
```suggestion
```

File: components/Main/Card.styled.js [hunk 11]
```diff
@@ -125,15 +124,22 @@ export const DeleteButton = styled.button`
...
+  font-size: 150%;
+  color: black;
 
   cursor: pointer;
 `;
 
 export const EditIcon = styled(MdEdit)`
-  font-size: 100%;
-  color: grey;
+  font-size: 150%;
```
Comment (onemanwenttomow) [inline components/Main/Card.styled.js]: I would consider using 1.5rem instead of 150% for the font-size to make it easier to understand and maintain.

**Review by onemanwenttomow — COMMENTED** [review]
@Leviron Nice job! 🎉

Added a few thoughts and suggestions.

**Comment by Leviron** [comment]
I took your suggestion and installed a useswr. it works. please have a look

File: components/Form/Edit.js [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+          type="text"
+          name="name"
+          placeholder="Recipe Name"
+          value={recipe.name}
+          onChange={handleInputChange}
+          required
+        />
+
+        {recipe.ingredients.map((ingredient, index) => (
+          <div key={index}>
```
Comment (Kheirah) [inline components/Form/Edit.js]: `index` is not enough for the `key` prop. You need a stable id or something else that is unique to your data.

**Review by Kheirah — APPROVED** [review]
Have a look at the key prop and see if you can use a name or something else instead.

**Pull request merged**